### ISSUE TITLE
Changing CB payment method does not work: 3D secure is not triggered

### ISSUE THREAD

**Issue by agenceKanvas** [woocommerce/woocommerce-gateway-stripe#3219]
**Describe the bug**
When a customer tries to change his payment method for another card, he enters his card details, validate but the 3D secure is never asked. The new payment is saved in WooCommerce but the next payment will not work because the 3D secure is not valid. The change is made using the new interface.

In stripe, we have the setup intent : 
![342287347-04f85c61-14a8-4463-8f5d-49fae5e83d8d](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/190149/37ed95f4-8d77-469b-842e-69e2c38b2ac9)

But it looks incomplete as the action is needed (probably the 3D secure).
![342279365-32f38354-b870-412a-9e4e-df97226ca548](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/190149/d3b568f7-e6ee-4d50-b761-2a482e6fc4fa)

**To Reproduce**
1. Login as a user with an active subscription
2. Try to change your payment method to a CB requiring a 3D Secure
3. Enter your CB info (In test mode, you can try with this card number (provided by Stripe) : 4000002760003184)

**Expected behavior**
The 3D Secure prompt must be triggered, and if valid, the payment method must be changed

**Environment (please complete the following information):**
 - WordPress Version : 6.5.4
 - WooCommerce Version : 8.9.3
 - Stripe Plugin Version : 8.0.1
 - Browser [e.g. chrome, safari] and Version : Chrome 

[SystemStatusReport_marsactu.fr_2024-06-24T10-23-41.txt](https://github.com/user-attachments/files/15953878/SystemStatusReport_marsactu.fr_2024-06-24T10-23-41.txt)

**Additional context**
I will let my colleagues add their versions but the bug is reproduced on other installs

**Comment by aheckler** [comment]
Sorta related: https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2305 I'm going to let our engineering team triage this one, although it does seem like a good enhancement to make.. :)

**Comment by superdaikon** [comment]
Hi. Same trouble here with subscription and 3DS action.  Our customers can't add a new card because the 3DS interface won't open. The fact that Stripe's return is in success (200) also means that the temporary payment method is included in the subscription. Customers don't see any errors, but of course all our payments then fail! 
Please check the code in detail
**setup_intent.requires_action** 
![shoot 2024-06-24 à 19 38 31](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/17338473/cfe1e4c3-9676-45fc-8238-43d0f35230fe)

Environment (please complete the following information):

    WordPress Version : 6.5.5
    WooCommerce Version : 8.9.3
    Stripe Plugin Version : 8.2.0
    Browser [e.g. chrome, safari] and Version : Firefox

**Comment by kevin25** [comment]
3D secure got stuck all the time. 
<img width="866" alt="Screen Shot 2024-06-25 at 09 24 20" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/2909982/bae51733-4dec-428b-9649-d71dd1fdfd11">

**Comment by agenceKanvas** [comment]
> Sorta related: #2305 I'm going to let our engineering team triage this one, although it does seem like a good enhancement to make.. :)

Thanks for your reply. Looks like more than just enhancement because a lot (and I mean a lot) of our customers will be lost because of that. And it will be the same for subscription based website

**Comment by agenceKanvas** [comment]
@aheckler 
Do you have some news about that ? We are a lot blocked by this, our customers do not receive the 3DS email so they can't update their subscription. But the source is still saved as functional but the future payment will of course not work.
Thanks !

**Comment by aheckler** [comment]
Tagging in @james-allan to triage as the current porter.

**Comment by agenceKanvas** [comment]
Thanks @aheckler then. I'll wait for @james-allan !

**Comment by james-allan** [comment]
Thanks for filing this issue @agenceKanvas and sorry for the delay in following up here.

While fixing a different issue (https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3307), I also fixed this issue. When a customer changes their subscription payment method, they will now be shown the 3DS popup modal. Prior to this change, the customer would have been redirected off site to complete the 3D-Secure challenge however that redirect would have been tripped up by off-site redirect protections. 

The customer now remains on the site and will complete the 3DS challenge like they would when making standard purchases. 

So just to recap, this issue has been fixed via https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3307 and the follow-up PR https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3312. These fixes will go out in our next release, scheduled for next week.

**Comment by agenceKanvas** [comment]
Hi @james-allan ! Thanks for your reply. I'm gonna wait for the udpate then.

**Comment by agenceKanvas** [comment]
I've tried the fix on my staging server and I have the 3DS modal. Awesome. I will apply the fix in production as it's a simple one. Thanks !

**Comment by kevin25** [comment]
> Thanks for filing this issue @agenceKanvas and sorry for the delay in following up here.
> 
> While fixing a different issue (#3307), I also fixed this issue. When a customer changes their subscription payment method, they will now be shown the 3DS popup modal. Prior to this change, the customer would have been redirected off site to complete the 3D-Secure challenge however that redirect would have been tripped up by off-site redirect protections.
> 
> The customer now remains on the site and will complete the 3DS challenge like they would when making standard purchases.
> 
> So just to recap, this issue has been fixed via #3307 and the follow-up PR #3312. These fixes will go out in our next release, scheduled for next week.

Not only the subscription but the issue with the normal products as well.
